### DISCUSSION TITLE
Behavior of `PolyData.point_normals` and `PolyData.cell_normals` when `Normals` exist

### DISCUSSION THREAD

**Discussion by whophil** [pyvista/pyvista#2728] [Ideas]
Right now, `PolyData.cell_normals` and `PolyData.point_normals` invoke `PolyData.compute_normals()`, with default settings, and then return the `Normals` array.

These properties are used downstream by methods such as `PolyData.plot_normals()` - with the effect that `plot_normals()` can only plot normals which are computed with the default settings of `compute_normals()`. However, I often want to call `compute_normals()` with a non-default flag, and then plot those normals.

One way around this would be if `PolyData.cell_normals` and `PolyData.point_normals` did something like this:
```python
@property
def point_normals(self):
    if 'Normals' in self.point_arrays:
        # return the existing normals
    else:
        # call `compute_normals` and return the normals
```

This would allow the normals to be computed externally, stored with the mesh, and then be picked up by the `cell_normals` or `point_normals` properties. Is there a good reason not to do this?

For my particular use case, adding a new argument `compute_normal_kwargs` to `PolyData.plot_normals()` would also work, but that seems uglier somehow.

**Comment by darikg** [comment]
+1 for this suggestion, I have found the current behavior surprising in the past
